### PR TITLE
improved error handling during retention policy setup

### PR DESCRIPTION
**Pull request by Excodibur**
Adresses https://github.com/ioBroker/ioBroker.influxdb/issues/140

### CODE REVIEW

**Review by Apollon77 — CHANGES_REQUESTED** [review]
Better change https://github.com/ioBroker/ioBroker.influxdb/pull/141/files#diff-5b47c21bdbb63faedda46ba27693786fb2e19157b6ab3da42bfeba083e1decd1R64 to a normal "err, value" callback and alwys have err first

and in successc case use (null, vale)

File: lib/DatabaseInfluxDB1x.js [hunk 1]
```diff
@@ -68,6 +68,10 @@ class DatabaseInfluxDB1x extends Database{
 
     async applyRetentionPolicyToDB(dbname, retention, callback) {
         this.getRetentionPolicyForDB(dbname, oldRetention => {
```
Comment (Apollon77) [inline lib/DatabaseInfluxDB1x.js:70]: ... and this becomes then

(err, oldRetention )=> {

File: lib/DatabaseInfluxDB1x.js [hunk 2]
```diff
@@ -61,19 +61,22 @@ class DatabaseInfluxDB1x extends Database{
...
             }
         });
     }
 
     async applyRetentionPolicyToDB(dbname, retention, callback) {
-        this.getRetentionPolicyForDB(dbname, oldRetention => {
+        this.getRetentionPolicyForDB(dbname, (err, oldRetention) => {
+            //Check, if there was an error
+            if (err)
+                callback(err);
```
Comment (Apollon77) [inline lib/DatabaseInfluxDB1x.js:73]: but I thiunk yoiu wantto "return" if error and not execute the other code or?!

**Review by Apollon77 — CHANGES_REQUESTED** [review]

File: lib/DatabaseInfluxDB1x.js (hunk 2 above)
Comment (Excodibur) [reply lib/DatabaseInfluxDB1x.js:73]: Added return back into it. You're right.

File: lib/DatabaseInfluxDB1x.js [hunk 3]
```diff
@@ -61,19 +61,24 @@ class DatabaseInfluxDB1x extends Database{
...
+                return;
+            }
+
             //Check, if it needs to be changed, otherwise skip
             this.log.debug("old retention: " + oldRetention + " new retention: " + retention);
             if ((oldRetention.time !== null) && (oldRetention.time === retention)) {
                 this.log.debug("Retention policy for " + dbname + " remains unchanged.");
                 callback(false);
-                return;
```
Comment (Apollon77) [inline lib/DatabaseInfluxDB1x.js:76]: but you also need that return or? Code needs to end here, right?

Comment (Excodibur) [reply lib/DatabaseInfluxDB1x.js]: true, added it. :-)

**Pull request merged**